File: programming files/datetime_convert.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
//bugs introduced by DD
public class datetime_convert {
    public static void main(String[] args) {
        
        // String that contains the date and time
        String dateStr = "2022-03-17 10:45:30";
        
        // Creates a formatter that will be used later
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        
        // Setting the dateObj object to the datetime using the formatter
        LocalDateTime dateObj = LocalDateTime.parse(dateStr, formatter);
        
        // Formatting the date and setting a varialbe equal to the result of the formatting
        String formattedDate = dateObj.format(DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss"));

        System.out.println(fomattedDate);
    }
}

File: programming files/datetime_convert.py
# bugs introduced: YM
from datetime import datetime

# String containing the date and time
date_str = "2022-03-17 %10:45:30"

# Getting the date from date_str and sets date_obj to the result
date_obj = datetime.strptime(date_str, '%Y-%m-%d H:%M:%S')

# Formats the date and sets formatted_date equal to the result
formatted_date = date_obj.strftime('%m/%d/%Y %H:%M:%S')

print(formatted_dat)
